src/components/item/form/DocumentForm.tsx

```diff
@@ -1,16 +1,88 @@
+import { FormControl, InputLabel, MenuItem, Select } from '@mui/material';
 import Box from '@mui/material/Box';
 import Typography from '@mui/material/Typography';
 
-import { DocumentItemType, ItemType } from '@graasp/sdk';
+import { useEffect, useState } from 'react';
+
+import {
+  DocumentItemExtraProperties,
+  DocumentItemType,
+  ItemType,
+} from '@graasp/sdk';
 import { DocumentItemTypeRecord } from '@graasp/sdk/frontend';
 import { BUILDER } from '@graasp/translations';
-import { TextEditor } from '@graasp/ui';
+import { DocumentItem } from '@graasp/ui';
 
 import { useBuilderTranslation } from '../../../config/i18n';
 import { ITEM_FORM_DOCUMENT_TEXT_ID } from '../../../config/selectors';
 import { buildDocumentExtra } from '../../../utils/itemExtra';
 import BaseForm from './BaseItemForm';
 
+export const DocumentExtraForm = ({
+  documentItemId,
+  extra,
+  maxHeight,
+  onCancel,
+  onContentChange,
+  onContentSave,
+  onFlavorChange,
+  placeholder,
+  saveButtonId,
+  showActions = false,
+}: {
+  documentItemId?: string;
+  extra: DocumentItemExtraProperties;
+  maxHeight?: string;
+  onCancel?: () => void;
+  onContentChange?: (text: string) => void;
+  onContentSave?: (text: string) => void;
+  onFlavorChange?: (text: DocumentItemExtraProperties['flavor']) => void;
+  placeholder?: string;
+  saveButtonId?: string;
+  showActions?: boolean;
+}): JSX.Element => (
+  <>
+    <Box sx={{ width: '100%' }}>
+      <FormControl variant="standard" sx={{ width: '50%', my: 1 }}>
+        <InputLabel shrink>Flavor</InputLabel>
+        <Select
+          variant="standard"
+          label="flavor"
+          value={extra.flavor}
+          onChange={(event) =>
+            onFlavorChange(
+              event.target.value === ''
+                ? undefined
+                : (event.target.value as DocumentItemExtraProperties['flavor']),
+            )
+          }
+        >
+          <MenuItem value="">None</MenuItem>
+          {['info', 'warning', 'error', 'success'].map((f) => (
+            <MenuItem key={f} value={f}>
+              {f}
```

Comment:
Should we add `i18n`? So we can also rename the flavors.

Reply:
Also should we create an enum for the flavor values? So we can also narrow the value in the backend for example.

Reply:
> Should we add `i18n`? So we can also rename the flavors.

Yes, I will add the PR to translations.

> Also should we create an enum for the flavor values? So we can also narrow the value in the backend for example.

This is one of those type vs enum thing: currently it is defined as an union type (see related PR: https://github.com/graasp/graasp-sdk/blob/2fdaddd15d48014215192af280b47ddbcef51b3c/src/interfaces/extra.ts#L26). Should I change it into an enum? @pyphilia @spaenleh 
Some fodder for argument:
- https://www.bam.tech/article/should-you-use-enums-or-union-types-in-typescript
- https://www.becomebetterprogrammer.com/typescript-union-types-vs-enums/
- https://stackoverflow.com/questions/40275832/typescript-has-unions-so-are-enums-redundant